internal/server/routes.go

```diff
@@ -141,6 +139,18 @@ func (s *Server) GenerateRoutes() Routes {
 	a.addPreviousVersionHandlersSignup()
 	a.addPreviousVersionHandlersGrants()
 
+	// Deprecated
+	add(a, authn, http.MethodPut, "/api/settings", route[api.EmptyRequest, *api.Settings]{
```

Comment:
Overall looks good! Is this required by older clients? If not we can do without a deprecated route